### PR TITLE
Add subcategories for Security

### PR DESCRIPTION
**Pull request by josemiguelmelo**

### CODE REVIEW

File: src/main/scala/codacy/codesniffer/docsgen/SubCategoriesMapper.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  def subcategoryFor(patternIdParts: PatternIdParts): Option[Pattern.Subcategory.Value] =
+    manualSubCategories
+      .get(patternIdParts.patternId.value)
+      .orElse(tryToGuessSubCategory(patternIdParts.prefix, patternIdParts.sniffType, patternIdParts.patternName))
+
+  private[this] def tryToGuessSubCategory(patternPrefix: String,
+                                          sniffType: String,
+                                          patternName: String): Option[Pattern.Subcategory.Value] = {
+
+    Option(sniffType.toLowerCase)
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/SubCategoriesMapper.scala]: Codacy Issue found: [Use pattern matching directly in function definitions](https://app.codacy.com/gh/codacy/codacy-codesniffer/pullRequest?prid=4955957)

File: src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala [hunk 2]
```diff
@@ -45,8 +45,9 @@ trait DocsParser {
       val category = CategoriesMapper.categoryFor(idParts, fallBackCategory)
 
       val spec = Pattern.Specification(idParts.patternId,
-                                       issueTypeFor(category, sourceFile, Result.Level.Info),
-                                       CategoriesMapper.categoryFor(idParts, fallBackCategory),
+                                       issueTypeFor(category._1, sourceFile, Result.Level.Info),
+                                       category._1,
+                                       category._2,
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/gh/codacy/codacy-codesniffer/pullRequest?prid=4955957)

File: src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala [hunk 3]
```diff
@@ -45,8 +45,9 @@ trait DocsParser {
       val category = CategoriesMapper.categoryFor(idParts, fallBackCategory)
 
       val spec = Pattern.Specification(idParts.patternId,
-                                       issueTypeFor(category, sourceFile, Result.Level.Info),
-                                       CategoriesMapper.categoryFor(idParts, fallBackCategory),
+                                       issueTypeFor(category._1, sourceFile, Result.Level.Info),
+                                       category._1,
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/gh/codacy/codacy-codesniffer/pullRequest?prid=4955957)

File: src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala [hunk 4]
```diff
@@ -45,8 +45,9 @@ trait DocsParser {
       val category = CategoriesMapper.categoryFor(idParts, fallBackCategory)
 
       val spec = Pattern.Specification(idParts.patternId,
-                                       issueTypeFor(category, sourceFile, Result.Level.Info),
-                                       CategoriesMapper.categoryFor(idParts, fallBackCategory),
+                                       issueTypeFor(category._1, sourceFile, Result.Level.Info),
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/gh/codacy/codacy-codesniffer/pullRequest?prid=4955957)

File: src/main/scala/codacy/codesniffer/docsgen/CategoriesMapper.scala [hunk 5]
```diff
@@ -1,26 +1,33 @@
...
-import com.codacy.plugins.api.results.Pattern.Category
+import com.codacy.plugins.api.results.Pattern.{Category, Subcategory}
 
 object CategoriesMapper {
 
-  private[this] val manualCategories: Map[String, Pattern.Category.Value] =
-    Map("WordPress_DB_PreparedSQL" -> Category.Security,
-        "WordPress_WP_PreparedSQL" -> Category.Security,
-        "WordPress_XSS_EscapeOutput" -> Category.Security)
+  def categoryFor(
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/CategoriesMapper.scala]: Codacy Issue found: [Cyclomatic complexity of 18 exceeds max of 10](https://app.codacy.com/gh/codacy/codacy-codesniffer/pullRequest?prid=4955957)

File: src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala (hunk 2 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/gh/codacy/codacy-codesniffer/pullRequest?prid=4955957)

File: src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala (hunk 3 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/gh/codacy/codacy-codesniffer/pullRequest?prid=4955957)

File: src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala (hunk 4 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/gh/codacy/codacy-codesniffer/pullRequest?prid=4955957)

File: src/main/scala/codacy/codesniffer/docsgen/CategoriesMapper.scala [hunk 6]
```diff
@@ -1,26 +1,34 @@
...
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/CategoriesMapper.scala]: Codacy Issue found: [Cyclomatic complexity of 18 exceeds max of 10](https://app.codacy.com/gh/codacy/codacy-codesniffer/pullRequest?prid=4955957)

Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/CategoriesMapper.scala]: Codacy Issue found: [Cyclomatic complexity of 18 exceeds max of 10](https://app.codacy.com/gh/codacy/codacy-codesniffer/pullRequest?prid=4955957)

File: project/plugins.sbt [hunk 7]
```diff
@@ -1,3 +1,3 @@
 resolvers += Resolver.jcenterRepo
 
-addSbtPlugin("com.codacy" % "codacy-sbt-plugin" % "17.1.5")
+addSbtPlugin("com.codacy" % "codacy-sbt-plugin" % "17.1.6")
```
Comment (lolgab) [inline project/plugins.sbt]: Here bump to latest version

File: src/main/resources/docs/description/Generic_Files_ExecutableFile.md [hunk 8]
```diff
@@ -0,0 +1 @@
+Files should not be executable.
```
Comment (lolgab) [inline src/main/resources/docs/description/Generic_Files_ExecutableFile.md]: Is this correct?

Comment (josemiguelmelo) [reply src/main/resources/docs/description/Generic_Files_ExecutableFile.md]: Yes, https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/Generic/Sniffs/Files/ExecutableFileSniff.php

File: project/plugins.sbt (hunk 7 above)
Comment (josemiguelmelo) [reply project/plugins.sbt]: Done ✅

**Pull request merged**